pkg/network/protocols/testutil/serverutils.go

```diff
@@ -0,0 +1,49 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package testutil
+
+import (
+	"os/exec"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+// RunDockerServer is a template for running a protocols server in a docker.
+// - serverName is a friendly name of the server we are setting (AMQP, mongo, etc.).
+// - dockerPath is the path for the docker-compose.
+// - env is any environment variable required for running the server.
+// - serverStartRegex is a regex to be matched on the server logs to ensure it started correctly.
+func RunDockerServer(t *testing.T, serverName, dockerPath string, env []string, serverStartRegex *regexp.Regexp) {
+	t.Helper()
+
+	cmd := exec.Command("docker-compose", "-f", dockerPath, "up")
+	patternScanner := NewScanner(serverStartRegex, make(chan struct{}, 1))
+
+	cmd.Stdout = patternScanner
+	cmd.Env = append(cmd.Env, env...)
+	go func() {
+		require.NoErrorf(t, cmd.Run(), "could not start %s with docker-compose", serverName)
+	}()
+
+	t.Cleanup(func() {
+		c := exec.Command("docker-compose", "-f", dockerPath, "down", "--remove-orphans")
+		c.Env = append(c.Env, env...)
+		_ = c.Run()
```

Comment:
why not checking an error of the `Run` command? it might save you the 30 seconds of waiting to get the proper log from the channel

Reply:
that's during the cleanup
if we fail to stop the dockers, we don't have much to do

Reply:
my bad i missed the cleanup scope